### ISSUE TITLE
JavaScript errors when running widget

### ISSUE THREAD

**Issue by FransVerschoor** [AuraQ/AutoCompleteForMendix#24]
Hi, thanks for the great widget. Highly appreciated. I have some remaining issues to be able to fully use it though. I have one issue that is reproducible with attached project: That is that on a NoResult I open a page to edit the object I want to create. When I Cancel that page, enter a new searchstring and click the No Results Found button nothing happens and I see the following error in the console:
```
Uncaught Error: Validate values using validator#validate(value, attr) before calling MxObject#set.
    at Object.set (mxui.js?636295085723576207:10)
    at HTMLAnchorElement.onclick (AutoCompleteForMendix.js:286)
```
When I first save the object, enter a searchstring and click the No Results Found button it does work.
In the console I also notice the following error which happens regularly:
```
Uncaught TypeError: ((jQuery.event.special[handleObj.origType] || (intermediate value)).handle || handleObj.handler).apply is not a function
    at HTMLDocument.dispatch (jquery-1.11.2.js:4665)
    at HTMLDocument.elemData.handle (jquery-1.11.2.js:4333)
```
I have a hunch that this is a cause for another issue which I cannot yet reproduce in this project but can in my actual project. In that case I need click the save on the new object page twice to actually save and close the page.
Hope you can help.
Frans
[TestAutoComplete.zip](https://github.com/AuraQ/AutoCompleteForMendix/files/976310/TestAutoComplete.zip)

**Comment by lindski** [comment]
Hi Frans

Looks like the first issue is caused by the cancel button rolling back the data view object (Travel) in the pop up which is affecting the widget's context object. The context object appears to still be pointing to the correct Travel object but its _guid property has been updated to "0". This "0" value is resulting in the error being thrown when the widget tries to set the 'Search String Attribute' value prior to calling the no results microflow after the cancel.

I don't think I have any control over this so our only option is to work around it.

The way I worked around it was to replace the default cancel button with an action button that calls a microflow to delete the Location object (we have cancelled so don't need it any more), empty the relevant association from the Travel object and then close the page. I also made sure that the close action on the page was pointing to my new action button.

I've attached an updated version of your test project that contains the workaround (rename the file to remove the .zip part)... let me know if this will work for you.
[TestAutoComplete_IL.mpk.zip](https://github.com/AuraQ/AutoCompleteForMendix/files/977045/TestAutoComplete_IL.mpk.zip)

Haven't seen the second error before... if you are able to reproduce it in a test project I'll be happy to take a look for you

Cheers

iain
